inc/az_ulib_ipc_api.h

```diff
@@ -292,13 +326,92 @@ AZ_NODISCARD az_result az_ulib_ipc_release_interface(az_ulib_ipc_interface_handl
     az_ulib_model_in model_in,
     az_ulib_model_out model_out);
 
+/**
+ * @brief   Synchronously Call a published procedure using string models.
+ *
+ * @param[in]   interface_handle    The #az_ulib_ipc_interface_handle with the interface handle.
+ *                                  It cannot be `NULL`. Call az_ulib_ipc_try_get_interface() to
+ *                                  get the interface handle.
+ * @param[in]   command_index       The #az_ulib_capability_index with the command index. Call
+ *                                  az_ulib_ipc_try_get_capability() to get the command index.
+ * @param[in]   model_in_span       The #az_span with the model in.
+ * @param[out]  model_out_span      The pointer to #az_span where the capability should store the
+ *                                  output content.
+ *
+ * @pre     IPC shall already been initialized.
+ * @pre     \p interface_handle shall be different than `NULL`.
+ *
+ * @return The #az_result with the result of the call.
+ *  @retval #AZ_OK                              If the IPC get success calling the procedure.
+ *  @retval #AZ_ERROR_ITEM_NOT_FOUND            If the target command does not exist.
+ *  @retval Others                              Defined by the target function.
+ */
+AZ_NODISCARD az_result az_ulib_ipc_call_w_str(
+    az_ulib_ipc_interface_handle interface_handle,
+    az_ulib_capability_index command_index,
+    az_span model_in_span,
+    az_span* model_out_span);
+
+/**
+ * @brief   Query IPC information.
+ *
+ * Creates a query for IPC. The query retrieves information from the IPC, depending on the content
+ * of the `query` argument. There are 2 valid query strings:
+ *
+ *  1) Empty string: query will return a list of all published interfaces.
+ *  2) Interface name: query will return a list of all capabilities in the interface.
+ *
+ * The result of the query will be a list with the information separated by comma.
+ *
+ * @param[in]   query               The `az_span` with the query string.
+ * @param[in]   result              The `az_span` with the buffer to return the query result.
+ * @param[in]   continuation_token  The pointer to `uint32_t` to return the query continuation
+ *                                  token. If the continuation token is 0, means that there is
+ *                                  no more information to return in this query, if it is
+ *                                  different than 0, means that a call to
+ *                                  az_ulib_ipc_query_next() shall return more information.
+ *
+ * @pre     IPC shall already been initialized.
+ * @pre     \p result shall be a valid az_span with at least 1 position.
+ * @pre     \p continuation_token shall be different than `NULL`.
+ *
+ * @return The #az_result with the result of the call.
+ *  @retval #AZ_OK                      If the succeeded and the result and continuation have
+ *                                      valid information.
```

Comment:
nit: If the query call succeeded and the result or continuation has valid information.

Reply:
same for query next. thanks!